testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/oauth/OAuthClient.java

```diff
@@ -298,157 +282,101 @@ public void doLoginGrant(String username, String password) {
-    public AccessTokenResponse doTokenExchange(String token, List<String> targetAudiences,
+    /**
+     * @deprecated Set clientId and clientSecret using {@link #client(String, String)} and use {@link #tokenExchangeRequest(String)}
+     */
+    @Deprecated
+    public AccessTokenResponse doTokenExchange(String subjectToken, List<String> targetAudiences,
                                                String clientId, String clientSecret, Map<String, String> additionalParams) throws Exception {
-        return new TokenExchangeRequest(token, clientId, clientSecret, this)
+        return tokenExchangeRequest(subjectToken)
+                .client(clientId, clientSecret)
                 .audience(targetAudiences)
                 .additionalParams(additionalParams).send();
     }
 
-    public JSONWebKeySet doCertsRequest() throws Exception {
-        HttpGet get = new HttpGet(getEndpoints().getJwks());
-        try (CloseableHttpResponse response = httpClientManager.get().execute(get)) {
-            return JsonSerialization.readValue(response.getEntity().getContent(), JSONWebKeySet.class);
-        }
+    public ClientCredentialsGrantRequest clientCredentialsGrantRequest() {
+        return new ClientCredentialsGrantRequest(this);
     }
 
-    public AccessTokenResponse doClientCredentialsGrantAccessTokenRequest(String clientSecret) throws Exception {
-        HttpPost post = new HttpPost(getEndpoints().getToken());
-
-        if (clientSecret != null) {
-            String authorization = BasicAuthHelper.RFC6749.createHeader(clientId, clientSecret);
-            post.setHeader("Authorization", authorization);
-        }
-        if (dpopProof != null) {
-            post.addHeader(TokenUtil.TOKEN_TYPE_DPOP, dpopProof);
-        }
-
-        List<NameValuePair> parameters = new LinkedList<>();
-        parameters.add(new BasicNameValuePair(OAuth2Constants.GRANT_TYPE, OAuth2Constants.CLIENT_CREDENTIALS));
-
-        String scopeParam = openid ? TokenUtil.attachOIDCScope(scope) : scope;
-        if (scopeParam != null && !scopeParam.isEmpty()) {
-            parameters.add(new BasicNameValuePair(OAuth2Constants.SCOPE, scopeParam));
-        }
-
-        UrlEncodedFormEntity formEntity = new UrlEncodedFormEntity(parameters, StandardCharsets.UTF_8);
-        post.setEntity(formEntity);
+    public AccessTokenResponse doClientCredentialsGrantAccessTokenRequest() throws Exception {
+        return clientCredentialsGrantRequest().send();
+    }
 
-        return new AccessTokenResponse(httpClientManager.get().execute(post));
+    /**
+     * @deprecated Set clientId and clientSecret using {@link #client(String, String)} and use {@link #doClientCredentialsGrantAccessTokenRequest()}
+     */
+    @Deprecated
+    public AccessTokenResponse doClientCredentialsGrantAccessTokenRequest(String clientSecret) throws Exception {
+        return clientCredentialsGrantRequest().client(config.getClientId(), clientSecret).send();
     }
 
+    // TODO Deprecate
```

Comment:
I suppose you can deprecate this now as it was refactored to the new `BackchannelAuthenticationTokenRequest`.